arrays/arrays.js

```diff
@@ -0,0 +1,116 @@
+function isArrayWithArray(a) {
+  return Array.isArray(a);
+}
+
+function isArrayWithInstanceOf(a) {
+  return a instanceof Array;
+}
+
+function range() {
+  var arr = [];
+  var i = 0;
+  if (arguments.length == 1) {
+    for (i = 0; i < arguments[0]; i++) {
+      arr.push(i);
+    }
+  } else {
+    if (arguments.length == 2) {
+      if (arguments[0] <= arguments[1]) {
+        for (i = arguments[0]; i < arguments[1]; i++) {
+          arr.push(i);
+        }
+      } else {
+        for (i = arguments[0]; i > arguments[1]; i--) {
+          arr.push(i);
+        }
+      }
+    } else {
+      if (arguments.length == 3) {
+        if (arguments[0] <= arguments[1] && arguments[2] > 0) {
+          for (i = arguments[0]; i < arguments[1]; i += arguments[2]) {
+            arr.push(i);
+          }
+        } else {
+          if (arguments[0] > arguments[1] && arguments[2] < 0) {
+            for (i = arguments[0]; i > arguments[1]; i += arguments[2]) {
+              arr.push(i);
+            }
+          } else {
+            if (arguments[0] == arguments[1] && arguments[2] == 0) {
+              arr.push(arguments[0]);
+            }
+          }
+        }
+      }
+    }
+  }
+  return arr;
+}
+
+function compactWithCycle(a) {
```

Comment:
could you please read task definition again?
compact should exclude all falsy values and not return new array like [true, true, true]. it's useless